src/snakemake/resources.py

```
@@ -382,8 +290,14 @@
             return True
 
     @classmethod
-    def _merge_resource_dict(cls, resources, skip=[], methods={}, default_method=max):
-        grouped = {}
+    def _merge_resource_dict(
+        cls,
+        resources: list[dict[str, str | int]],
+        skip: list[str] = [],
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)